packages/nl-design-system-unstable/__docs__/form-patterns.md

```diff
@@ -1,139 +1,270 @@
+
+### Help en feedback tekst
+
+Optionele help of feedback tekst kan dynamisch zijn, bijvoorbeeld wanneer een veld onjuist is ingevuld en je na validatie wilt uitleggen hoe deze fout gecorrigeerd kan worden.
+
+Overweeg om extra help tekst toe te voegen wanneer:
+
+* Je een bezoeker vraagt om minder bekende informatie.
+* Een bezoeker zich af zou kunnen vragen waarom om specifieke informatie gevraagd wordt.
+* Een bezoeker mogelijk bezorgd kan zijn over de veiligheid of privacy van bepaalde gegevens.
+
+### Optionele en verplichte invoervelden
+
+Overweeg om optionele vragen aan het einde van het formulier te stellen, tenzij deze echt bij een bepaald onderwerp/groep horen.
+
+Als niet alle velden verplicht zijn, maak dan duidelijk welke velden dat wel zijn.
+
+Als de meeste invoer op een formulier verplicht is, geef dan enkel diegene aan die optioneel zijn. Als de meeste invoer op een formulier optioneel is, geef dan enkel diegene aan die verplicht zijn.
+
+De beste manier om dit aan te geven is met een tekst zodat deze ook door schermlezers wordt aangekondigd. Zijn de meeste velden vereist in een formulier? Laat dan alleen een indicatie zien voor de optionele velden. Zorg er echter voor dat dit patroon consistent is in de hele interface.
+
+De duidelijkste manier is door aan het tekstlabel van het betreffende element respectievelijk de tekst ‘(optioneel)’ of ‘(verplicht)’ toe te voegen. Dit zorgt er voor dat voorleessoftware deze makkelijk kan interpreteren. Eventueel kan een asterisk (*) geplaatst worden bij verplichte velden. Zorg in dat geval echter wel dat ook voorleessoftware deze kan interpreteren.
+
+## Validatie
+
+Een invoerelement kan gevalideerd worden als de focus van het element af is. Hierdoor is het mogelijk om direct een foutmelding te tonen als bijvoorbeeld het ingevoerde formaat niet voldoet aan de eisen voor dat veld. Als zodanig ziet een gebruiker de fout direct in context en kan deze dan onmiddellijk corrigeren.
+
+Feedback zou alleen op onjuiste velden weergegeven moeten zijn om de mentale belasting van bezoeker minimaal te houden. Tevens correcte velden aanduiden kan zorgen voor teveel visuele afleiding.
+
+Niet alle velden dienen gevalideerd te worden; voor een voornaam kunnen we uiteraard niet bepalen of deze al dan niet correct is.
+
+Bij het verzenden van een formulier kan echter wel gevalideerd worden of er daadwerkelijk iets in een veld ingevuld is.
+
+Een bezoeker zou niet gedwongen moeten worden een formulier volgens de visuele orde op het scherm in te vullen. Het moet bijvoorbeeld mogelijk zijn om eerst een achternaam-veld in te vullen voordat het voornaam-veld wordt ingevuld, zonder dat dit als een fout aangeduid wordt.
+
+## Overzicht en samenvatting
+
+Geef de gebruiker een overzicht van de ingevoerde informatie en zorg dat deze te wijzigen is.
```

Comment:
"zorg dat deze te wijzigen is" kan ambigu zijn: moet het overzicht ook weer bestaan uit form inputs? Als we zeggen "zodat het nog gecorrigeerd kan worden" of iets dergelijks hint misschien meer naar dat je terug kunt gaan naar een eerdere stap.

Overigens: is de samenvatting alleen voor meerstapsformulieren, of ook voor kleine formulieren?

Reply:
Dit moet inderdaad nog nader uitgewerkt worden.

V.w.b. samenvatting; hier is inderdaad wellicht onderscheid te maken tussen meerstappen en kleine formulieren.